### ISSUE TITLE
`SCANVI.loss` fails when `n_samples>1`

### ISSUE THREAD

**Issue by martinkim0** [scverse/scvi-tools#2429]
Reproducible script:
```
vae = SCANVAE(n_input=100, n_labels=1)
x = torch.randint(0, 100, (10, 100), dtype=torch.float32)
batch = torch.zeros(10, dtype=torch.long)
labels = torch.zeros(10, dtype=torch.long)
tensors = {
    REGISTRY_KEYS.X_KEY: x,
    REGISTRY_KEYS.BATCH_KEY: batch,
    REGISTRY_KEYS.LABELS_KEY: labels,
}

_ = vae.forward(
    tensors,
    inference_kwargs={"n_samples": 2}
)
```

**Comment by martinkim0** [comment]
sanity check with the [Atlas-level integration of lung data](https://docs.scvi-tools.org/en/stable/tutorials/notebooks/scrna/harmonization.html) tutorial: [harmonization.pdf](https://github.com/scverse/scvi-tools/files/14829605/harmonization.pdf)

**Comment by ori-kron-wis** [comment]
see https://github.com/scverse/scvi-tools/pull/2902 with the fix
